### ISSUE TITLE
Nit: Tweak Display Template

### ISSUE THREAD

**Issue by tvraman** [isamert/empv.el#45]
Suggestion For Info Line Display

The new template is nice, I can now put the title in front.

There is still the minor annoyance of HLS live streams displaying a
somewhat irrelevant chapter and percentage, best shown with an
example.

Stream: BBC Radio 4
Displays:  [radio-4-bbc.m3u Playing, 00:06 of 00:19 (34%), 1/1, 🔊 50]
m3u file below:

#EXTINF:-1,BBC - Radio 4
http://as-hls-ww-live.akamaized.net/pool_904/live/ww/bbc_radio_fourfm/bbc_radio_fourfm.isml/bbc_radio_fourfm-audio%3d96000.norewind.m3u8

**Comment by isamert** [comment]
What's the value of your `empv-display-current-format`? And which part in

>  [radio-4-bbc.m3u Playing, 00:06 of 00:19 (34%), 1/1, 🔊 50]

you find that irrelevant? Simply excluding `#{chapter}` from the template should get rid of the chapter information and I don't see the chapter information in the example you gave. Am I missing something?

**Comment by tvraman** [comment]
Current value:
empv-display-current-format'
"[#{title} #{volume} #{state}#{item-loop-indicator}, #{time-pos} of
#{duration} (#{percent-pos}%),
#{playlist-pos}/#{playlist-count}#{playlist-loop-indicator}#{radio}#{speed}]
#{chapter}"

Note that my playlist has only one entry --
what you see displayed
[radio-4-bbc.m3u , 🔊 125 Playing, 00:05 of 00:19 (26%), 1/1]
the 00:05 of 00:19 is the current fragment they're streaming; HLS
basically streams the live streams as a sequence of small segments.
--

**Comment by isamert** [comment]
What would be the solution for this? Should `#{time-pos}` `${duration}` and `${percent-pos}` evaluate to empty string if it's an HLS fragment?

Sorry for the late reply.

**Comment by tvraman** [comment]
İsa Mert Gürbüz ***@***.***> writes:

That would be one possibility since those hls fragment numbers and
duration values are irrelevant to the user.
> What would be the solution for this? Should #{time-pos} ${duration}
> and ${percent-pos} evaluate to empty string if it's an HLS fragment?
>
> Sorry for the late reply.
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.
>

--

**Comment by isamert** [comment]
I thought about this a little bit and found out that evaluating some values to empty string in case of HLS format causes more problems than it fixes. So I went ahead and changed how `empv-display-current-format` works. Now, it can be an alist from file formats to display template so that you can change the template based on file format if you want. Let me know if it works for you!

**Comment by tvraman** [comment]
thanks will give it a try.
--
